src/java.base/share/classes/jdk/internal/vm/annotation/IntrinsicCandidate.java

```diff
@@ -28,94 +28,67 @@
- * should consider buffering them in an array. Removing field access from intrinsics
- * not only clarifies the interface with between the JVM and JDK; it also helps decouple
- * the HotSpot and JDK implementations, since if JDK code before and after the intrinsic
- * manages all field accesses, then intrinsics can be coded to be agnostic of object
- * layouts.</li>
- *
- * Maintainers of the HotSpot VM must consider the following when modifying
- * intrinsics.
- *
- * <ul>
- * <li>When adding a new intrinsic, make sure that the corresponding method
- * in the Java libraries is annotated with {@code @IntrinsicCandidate}
- * and that all possible call sequences that result in calling the intrinsic contain
- * the checks omitted by the intrinsic (if any).</li>
- * <li>When modifying an existing intrinsic, the Java libraries must be updated
- * to match the semantics of the intrinsic and to execute all checks omitted
- * by the intrinsic (if any).</li>
- * </ul>
- *
- * Persons not directly involved with maintaining the Java libraries or the
- * HotSpot VM can safely ignore the fact that a method is annotated with
- * {@code @IntrinsicCandidate}.
- *
- * The HotSpot VM defines (internally) a list of intrinsics. Not all intrinsic
- * are available on all platforms supported by the HotSpot VM. Furthermore,
- * the availability of an intrinsic on a given platform depends on the
- * configuration of the HotSpot VM (e.g., the set of VM flags enabled).
- * Therefore, annotating a method with {@code @IntrinsicCandidate} does
- * not guarantee that the marked method is intrinsified by the HotSpot VM.
- *
- * If the {@code CheckIntrinsics} VM flag is enabled, the HotSpot VM checks
- * (when loading a class) that (1) all methods of that class that are also on
- * the VM's list of intrinsics are annotated with {@code @IntrinsicCandidate}
- * and that (2) for all methods of that class annotated with
- * {@code @IntrinsicCandidate} there is an intrinsic in the list.
+ * <h2 id="intrinsification">Intrinsification</h2>
+ * The most frequently special treatment is intrinsification, which replaces a
+ * candidate method's body, bytecode or native, with handwritten platform
+ * assembly and/or compiler IR.  (See {@code LibraryCallKit::try_to_inline} in
+ * {@code library_call.cpp} for logic that checks if an intrinsic is available
+ * and applicable at a given call site.)  Many Java library methods have
+ * properties that cannot be deduced by the compiler for optimization, or can
+ * utilize specific hardware instructions not modeled by the compiler IR, making
+ * intrinsics necessary.
+ * <p>
+ * Intrinsification may never happen, or happen at any moment during execution.
+ * For example, the bytecodes of a candidate method may be executed by lower
+ * compilation tiers of VM execution, while higher compilation tiers may replace
+ * the bytecodes with specialized assembly code and/or compiler IR.  Therefore,
```

Comment:
> while higher compilation tiers may replace the bytecodes with specialized assembly code and/or compiler IR

Is there ever a case, where for a `@IntrinsicCandidate` method, the runtime will choose to execute the instrinsic for that method for a certain duration and then at a later point in time replace the intrinsic with compiler generated code? In other words, once the runtime executes the intrinsic implementation for a `@IntrinsicCandidate` method, will the method's implementation be switched to anything else during the lifetime of an application?

Reply:
We cannot rule it out, but this sentence begins "for example" meaning this is just one scenario and is not exhaustive.

Reply:
To address your concern, I have reworded:
```
 * During execution, intrinsification may happen and may be rolled back at any
 * moment; this loading and unloading process may happen zero to many times.
```